### PR TITLE
Cluster setup article and more

### PR DESCRIPTION
**Pull request by sibiryakov**

### CODE REVIEW

File: docs/source/topics/cluster-setup.rst [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        'frontera.contrib.scrapy.middlewares.seeds.file.FileSeedLoader': 1,
+    })
+    DOWNLOADER_MIDDLEWARES.update({
+        'frontera.contrib.scrapy.middlewares.schedulers.SchedulerDownloaderMiddleware': 999,
+    })
+
+
+Starting the cluster
+====================
+
```
Comment (voith) [inline docs/source/topics/cluster-setup.rst]: I think there's a note needed here saying that frontera is sensitive to the order in which the components are started. First start DBW, followed by SW and than the spiders.

File: docs/source/topics/dns-service.rst [hunk 2]
```diff
@@ -0,0 +1,21 @@
+===========
+DNS Service
+===========
+
```
Comment (voith) [inline docs/source/topics/dns-service.rst]: May be a tutroial for writing in memory cache like: https://github.com/sibiryakov/frontera-google/blob/master/bc/broadcrawl/__init__.py#L14 and how it can be used with the crawling strategy will be really helpful. I think adding https://github.com/sibiryakov/frontera-google to examples will be really helpful for new comers.

File: docs/source/topics/fine-tuning.rst [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+* :term:`strategy worker` performance,
+* :term:`db worker` batch generation insufficiency.
+* HBase response times are too high,
+* Network within cluster is overloaded.
+
+Tuning HBase
+============
+* Increase block cache in HBase.
+* Put Thrift server on each HBase region server and spread load from SW to Thrift.
+* Enable Snappy compression.
```
Comment (voith) [inline docs/source/topics/fine-tuning.rst]: may be we can add (see :setting:`HBASE_USE_SNAPPY`))

File: docs/source/topics/fine-tuning.rst [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+Flow control between various components
+=======================================
+
+The :setting:`MAX_NEXT_REQUESTS` is used for controlling the batch size. In spiders config it controls how much items
+will be consumed per one :attr:`get_next_requests <frontera.core.manager.FrontierManager.get_next_requests>` call. At
+the same time in DB worker config it sets count of items to generate per partition. When setting these parameters keep
+in mind:
+
+* DB worker and spider values have to be consistent to avoid overloading of message bus and loosing messages. In other
+  words, DB worker have to produce slightly more than consumed by spiders, because the spider should still be able to
```
Comment (voith) [inline docs/source/topics/fine-tuning.rst]: This sentence is a little confusing to me. Did you mean:
In other  words, DB worker has to produce slightly more messages than those consumed by spiders. 

<del>because the spider should still be able to fetch new pages even though the DB worker has not pushed a new batch yet</del> Spiders should be able to fetch new pages, even if the DB worker has not pushed in a new batch.

My English is bad, but this is what seems right to me :)

File: docs/source/topics/fine-tuning.rst [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  fetch new pages even though the DB worker has not pushed a new batch yet.
+* Spider consumption rate is depending on many factors: internet connection latency, amount of spider
```
Comment (voith) [inline docs/source/topics/fine-tuning.rst]: is depending --> depends

File: docs/source/topics/loggers.rst [hunk 6]
```diff
@@ -0,0 +1,34 @@
+=======
+Logging
+=======
+
+Frontera is using Python native logging system. That means you can manage what messages are logged and where by writing
```
Comment (voith) [inline docs/source/topics/loggers.rst]: I would rephrase this as :
Frontera uses Python native logging system. This allows a user to manage logged messages by writing a logger configuration file (see :setting:`LOGGING_CONFIG`).

File: docs/source/topics/cluster-setup.rst [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+Things to setup before you start
+================================
+* Kafka,
+* HBase (we recommend 1.0.x and higher),
+* :doc:`DNS Service <dns-service>` (recommended but not required).
+
+Things to implement before you start
+====================================
+* :doc:`Crawling strategy <own_crawling_strategy>`
```
Comment (sibiryakov) [inline docs/source/topics/cluster-setup.rst]: Sure, you're right!

File: docs/source/topics/dns-service.rst (hunk 2 above)
Comment (sibiryakov) [inline docs/source/topics/dns-service.rst]: Good point.

File: docs/source/topics/fine-tuning.rst (hunk 4 above)
Comment (sibiryakov) [inline docs/source/topics/fine-tuning.rst]: This sentence describe an event when some data needs to be available even if DBW can't keep up for spiders. So it should be tuned in a way to slightly overcome the spiders consumption.
